### DISCUSSION TITLE
[Feature] Memories for shared album members

### DISCUSSION THREAD

**Discussion by SteffenAuer** [immich-app/immich#3780] [Feature Request]
## Feature

I think memories are a great addition. I’m sort of curious every day what happened in the past years xD.
What would be super neat imo would be if memories would also be created for the members of shared albums. So say user A shares an album with user B. Then, of course, user A sees the memory with those photos. I think it would be great if user B’s memories would also include the photos of that same shared album.

## Platforms
- [x] Server
- [ ] Web
- [ ] Mobile

**Comment by heliohenriques** [comment]
Hi,
I also would like to see this feature. 
It was great feature to add on "sharing with partners".

**Comment by themarfman** [comment]
This is also a great feature idea when folks are primarily using external libraries.  For example, my account manages the external library.  I use the share with partner feature to share with my wife's account.  However, she doesn't see memories for the external library.

**Comment by Sciamano72** [comment]
I also would love to have memories on Shared with Partners libraries.

**Comment by StefanRos** [comment]
I would be happy about that too.

**Reply by heliohenriques** [reply]
This was implemented while ago.
The user only need to go to settings and enable that option on "Partner Sharing".
You can choose how many partners you want. 
Note: if you want to see your partner photos, she need to do the same.

**Reply by StefanRos** [reply]
But the partner does not receive any image memories from the partner library.

**Reply by heliohenriques** [reply]
Check your settings.
![Screenshot](https://github.com/user-attachments/assets/ddfdceea-e4e1-4fbe-8c72-b8551aed3859)

**Reply by StefanRos** [reply]
This is the partner's view. Everything appears to be working, but no memories are present.
![image](https://github.com/user-attachments/assets/dcbf50a4-4f6f-4173-9d4f-d2e8a6b6727a)

**Reply by heliohenriques** [reply]
You are not sharing your photos with your partner.
Only she's sharing with you.

**Reply by acidicX** [reply]
@heliohenriques nope, exactly the same on my end (with 1.134.0). My photos do appear in my partners timeline, but no memories are present for my partner to view. Memories are only available for me, so it's either broken or not implemented.

**Reply by alextran1502** [reply]
@acidicX can you try going to the jobs page, then on the upper top right, select create job and select memory generation?

**Reply by StefanRos** [reply]
Top, that works for me. It can really be that simple.

**Reply by tricktux** [reply]
Hi, I am facing the same issue. And the memory generation didn't do it for me 😭 .

Nothing happens when I create that job.

I've tried running back to back the `Memory Clean Up` and the `Memory Generation`. Would love this feature.

Thanks for the support.

**Reply by gbil** [reply]
Adding my vote here too, sharing my photos with my partner, memories work for me, they see nothing. I've enabled the relevant option in their preferences, tried with the job creation etc. nothing.

**Reply by acidicX** [reply]
@alextran1502 sadly nothing happens when I create that job

**Reply by darianferrer** [reply]
It worked for me and my partner. My account owns the external library which I've shared with my wife. Initially, she didn't get to see the memories while it worked fine in my account. I change her account to be admin and then when she ran the `Memory generation` she got it working; I think this is the important bit: the user missing the memories is the one that needs to run the job.

**Reply by acidicX** [reply]
Did also only work for me once I've made the other account admin and run the job in that account. This needs a bit of tweaking I'd say ;)

**Reply by Str1atum** [reply]
I agree. In the latest update the memory generation seems to work only for own photos. Even when sharing they are not used to generate memories - that makes a family library even more difficult to achieve.

**Comment by mybropro** [comment]
I'm on version 3.1.0 and this still does not seem to work. Wife is sharing with me and I (admin account) with her. We can see each other's images fine, but memories are only sourced from our own libraries. I tried re-running the mentioned jobs but that didn't seem to do anything.

**Reply by Str1atum** [reply]
Yes, unfortunately not working for months because of a design decision. Guess we have to wait for the new sharing mechanism.
